data_steward/test/unit_test/vocabulary_test.py

```diff
@@ -0,0 +1,97 @@
+import unittest
+import StringIO
+from vocabulary import *
+from vocabulary import _transform_csv
+import test_util
+import os
+import mock
+
+
+class VocabularyTest(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        print('**************************************************************')
+        print(cls.__name__)
+        print('**************************************************************')
+
+    def setUp(self):
+        super(VocabularyTest, self).setUp()
+
+    @staticmethod
+    def do_transform_file(input_text):
+        in_fp = StringIO.StringIO(input_text)
+        out_fp = StringIO.StringIO()
+        _transform_csv(in_fp, out_fp)
+        return out_fp.getvalue()
+
+    def test_transform_file(self):
+        header = ['concept_id', 'valid_start_date', 'valid_end_date', 'invalid_reason']
+        row1 = ['1', '2017-05-17', '2099-12-31', '']
+        row2 = ['2', '2019-01-01', '2099-12-31', '']
+        rows = [header, row1, row2]
+        lines = map(lambda r: DELIMITER.join(r), rows)
+
+        # after transform should output valid csv with date format yyyy-mm-dd
+        expected_text = LINE_TERMINATOR.join(lines) + LINE_TERMINATOR
+        input_text = expected_text.replace('-', '')
+        actual_text = VocabularyTest.do_transform_file(input_text)
+        self.assertEqual(expected_text, actual_text)
+
+        # windows line endings are replaced with linux ones
+        input_text = expected_text.replace(LINE_TERMINATOR, '\r\n')
+        actual_text = VocabularyTest.do_transform_file(input_text)
+        self.assertEqual(expected_text, actual_text)
+
+    def test_format_date_str(self):
+        self.assertEqual('2019-01-23', format_date_str('20190123'))
+        self.assertEqual('2019-01-23', format_date_str('2019-01-23'))
+        with self.assertRaises(ValueError):
+            format_date_str('201901234')
+
+    def test_append_vocabulary(self):
+        in_path = test_util.TEST_VOCABULARY_VOCABULARY_CSV
+        out_path = os.tempnam()
```

Comment:
What are the benefits of using `os.tempnam()` over `tempfile.mkdtemp()`?  I haven't run into this function before.

Also, you should wrap this whole thing in a `try/except/finally` block.  In the `finally` block remove everything you've made there.  That way, if this fails due to an assertion or for any other reason, those files get cleaned up.

Reply:
Thanks! I forgot about that library. I changed it to use `tempfile.mkdtemp()`. Does this try/finally block look ok?

Reply:
I feel better about it removing the files when the test is done.  From experience, it's a pain tracking down which failing test is littering your code base with temporary files.

Reply:
@lrwb-aou do you mean that you'd prefer for [this](https://github.com/mark-velez/curation/blob/add_gen_vocab/data_steward/test/unit_test/vocabulary_test.py#L115) to go to `tearDown()`?

Reply:
No, I don't.  You may not have files to remove every time a function executes.  I was just noticing the assert statements were in a context manager.  I think if something fails inside there, the remove statements execute because they are outside the context manager, but I haven't tried it locally to verify it.   I do know the `try/except/finally` block, will work every time.  This should work.  I was just saying I've done it wrong in the past and tried to find which test was littering my work environment with temp files.  That's a pain.  What you've done should work.